lib/danger/ci_source/gitlab_ci.rb

```diff
@@ -54,16 +61,29 @@ def self.determine_merge_request_id(env)
 
     def initialize(env)
       @env = env
-      @repo_slug = env["CI_MERGE_REQUEST_PROJECT_PATH"] || env["CI_PROJECT_PATH"]
-      @project_url = env["CI_MERGE_REQUEST_PROJECT_URL"] || env["CI_PROJECT_URL"]
+      @repo_slug = slug_from(env)
+      @project_url = env["DANGER_PROJECT_REPO_URL"] || env["CI_MERGE_REQUEST_PROJECT_URL"] || env["CI_PROJECT_URL"]
     end
 
     def supported_request_sources
-      @supported_request_sources ||= [Danger::RequestSources::GitLab]
+      @supported_request_sources ||= [
+        Danger::RequestSources::GitHub,
+        Danger::RequestSources::GitLab
+      ]
     end
 
     def pull_request_id
-      @pull_request_id ||= self.class.determine_merge_request_id(@env)
+      @pull_request_id ||= self.class.determine_pull_or_merge_request_id(@env)
+    end
+
+    private
+
+    def slug_from(env)
+      if env["DANGER_PROJECT_REPO_URL"]
+        env["DANGER_PROJECT_REPO_URL"].split('/').last(2).join('/')
```

Comment:
gitlab doesn't seem to give us any env var that points to the external repo's URL. It only points to gitlab's mirror URL. Users will need to set this ENV var themselves